be/src/main/java/kr/codesquad/sidedish/controller/ProductController.java

```diff
@@ -2,102 +2,117 @@
-	}
-
-	private CommonResponse noContentCommonResponse() {
-		return new CommonResponse(CommonCode.SUCCESS_NO_CONTENT);
-	}
-
-	private CommonResponse OKCommonResponse(Object data) {
-		return new CommonResponse(CommonCode.SUCCESS, data);
-	}
-
-	private List<ProductDTO> loadListByType(String dishType, Optional<String> sideDishType) {
-		if (sideDishType.isPresent()) {
-			return productService.loadSideDishListByType(DishType.stringToEnum(dishType),
-				SideDishType.stringToEnum(sideDishType.get()));
-		}
-		return productService.loadDishListByType(DishType.stringToEnum(dishType));
-	}
-
-	private void checkForExistingId(Integer id) {
-		if (id < MinimumProductId ||
-			id > MaximumProductId) {
-			throw new CustomException(ErrorCode.PRODUCT_ID_NOT_ALLOWED);
-		}
-	}
+    public static final Integer MinimumProductId = 1;
+    public static final Integer MaximumProductId = 24;
+
+    private final ProductService productService;
+    private final ShippingInfoService shippingInfoService;
+
+    @InitBinder
+    private void initBinder(final WebDataBinder webdataBinder) {
+        webdataBinder.registerCustomEditor(DishType.class, new PropertyEditorSupport() {
+            @Override
+            public void setAsText(String text) {
+                setValue(DishType.fromName(text));
+            }
+        });
+        webdataBinder.registerCustomEditor(SideDishType.class, new PropertyEditorSupport() {
+            @Override
+            public void setAsText(String text) {
+                setValue(SideDishType.fromName(text));
+            }
+        });
+    }
+
+    /**
+     * 요리 카테고리별 목록 불러오기
+     */
+    @ResponseBody
+    @GetMapping("/{dishType}")
+    public ResponseEntity<CommonResponse> loadDishProducts(@PathVariable DishType dishType) {
+        List<ProductDTO> productDTOs = productService.loadDishListByType(dishType);
+
+        if (productDTOs.size() == 0) {
+            return noContentCommonResponse().toResponseEntity();
+        }
+
+        List<ResponseSimpleProductInfo> simpleDTOs = productDTOs.stream()
+                .map(p -> ResponseSimpleProductInfo.from(p))
+                .collect(Collectors.toList());
+
+        return okCommonResponse(simpleDTOs).toResponseEntity();
+    }
+
+    /**
+     * 반찬 카테고리별 목록 불러오기
+     */
+    @ResponseBody
+    @GetMapping("/{dishType}/{sideDishType}")
+    public ResponseEntity<CommonResponse> loadSideDishProducts(@PathVariable DishType dishType, @PathVariable SideDishType sideDishType) {
+        List<ProductDTO> productDTOs = productService.loadSideDishListByType(dishType, sideDishType);
+
+        if (productDTOs.size() == 0) {
+            return noContentCommonResponse().toResponseEntity();
+        }
+
+        List<ResponseSimpleProductInfo> simpleDTOs = productDTOs.stream()
+                .map(p -> ResponseSimpleProductInfo.from(p))
+                .collect(Collectors.toList());
+
+        return okCommonResponse(simpleDTOs).toResponseEntity();
```

Comment:
```suggestion
        final List<ResponseSimpleProductInfo> simpleDtos = productService.loadSideDishListByType(dishType, sideDishType).stream()
            .map(ResponseSimpleProductInfo::from)
            .collect(Collectors.toList());

        return okCommonResponse(simpleDtos).toResponseEntity();
```

저라면 200, 204 구분 안하고 그냥 이렇게 만들 것 같아요. 

---

이전 질문에 이어서 생각해보니까

- `CommonResponse()` 생성할 때 들어온 body가 null이면 
- `204 CommonResponse()`을 만들게 할 수 도 있을 것 같습니다.

이렇게 하면 CommonResponse를 만드는 곳에서는 내용이 비어있는지 아닌지 체크할 필요도 없겠죠.

Reply:
생성할 때 body가 null 일 때 구분하는 것은 이해가 되었습니다!

> 저라면 200, 204 구분 안하고 그냥 이렇게 만들 것 같아요.

이 부분에 대한 이유를 알 수 있을까요? 서버 입장에서 요청에 대한 응답은 정상적으로 처리 된 것이라 굳이 구분할 필요가 없는건가요?

Reply:
아래 질문에 남긴 코멘트를 참고해주세요.

https://github.com/codesquad-members-2022/sidedish/pull/175#discussion_r861500358